crypto/keys/generate_test.go

```diff
@@ -98,16 +100,19 @@ func TestNewFromSpec(t *testing.T) {
 			wantErr: true,
 		},
 	} {
-		key, err := NewFromSpec(test.keySpec)
-		if gotErr := err != nil; gotErr != test.wantErr {
-			t.Errorf("%v: NewFromSpec() = (_, %v), want err? %v", test.desc, err, test.wantErr)
-			continue
-		} else if gotErr {
-			continue
-		}
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			key, err := NewFromSpec(test.keySpec)
```

Comment:
I think this will fall foul of https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables.